proofs.go

```diff
@@ -290,3 +433,49 @@ func DecompactProof(proof *SparseCompactMerkleProof, spec *TreeSpec) (*SparseMer
 		SiblingData:           proof.SiblingData,
 	}, nil
 }
+
+// CompactClosestProof compacts a proof, to reduce its size.
+func CompactClosestProof(proof *SparseMerkleClosestProof, spec *TreeSpec) (*SparseCompactMerkleClosestProof, error) {
+	if !proof.sanityCheck(spec) {
+		return nil, ErrBadProof
+	}
+	compactedProof, err := CompactProof(proof.ClosestProof, spec)
+	if err != nil {
+		return nil, err
+	}
+	flippedBits := make([]byte, len(proof.FlippedBits))
+	for i, v := range proof.FlippedBits {
+		flippedBits[i] = byte(v)
+	}
+	return &SparseCompactMerkleClosestProof{
+		Path:             proof.Path,
+		FlippedBits:      flippedBits,
+		Depth:            byte(proof.Depth),
```

Comment:
so the way this works is because the tree has a max depth of 256 aka 1 byte, I will be adding tests for this over the weekend but i am pretty sure its fine (famous last words)